src/Alert.php

```diff
@@ -4,269 +4,358 @@
+     *
+     * @param string $value The template content string.
+     *
+     * @return self A new instance of the current class with the specified template content.
      */
-    public function warning(): self
+    public function templateContent(string $value): self
     {
-        return $this->addClassNames('alert-warning');
+        $new = clone $this;
+        $new->templateContent = $value;
+
+        return $new;
     }
 
     /**
-     * Short method for info alert type
+     * Sets the HTML attributes for the toggle component in the alert.
+     *
+     * @param array $value Attribute values indexed by attribute names.
+     *
+     * @return self A new instance of the current class with the specified toggle attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function info(): self
+    public function toggleAttributes(array $value): self
     {
-        return $this->addClassNames('alert-info');
+        $new = clone $this;
+        $new->toggleAttributes = $value;
+
+        return $new;
     }
 
     /**
-     * Short method for light alert type
+     * Sets the toggle component to be rendered as a link instead of a button.
+     *
+     * @return self A new instance of the current class with the toggle set as a link.
      */
-    public function light(): self
+    public function toggleLink(): self
     {
-        return $this->addClassNames('alert-light');
+        $new = clone $this;
+        $new->toggleLink = true;
+
+        return $new;
     }
 
     /**
-     * Short method for dark alert type
+     * Set the alert type. The following options are allowed:
+     * - `Type::PRIMARY`: Primary alert.
+     * - `Type::SECONDARY`: Secondary alert.
+     * - `Type::SUCCESS`: Success alert.
+     * - `Type::DANGER`: Danger alert.
+     * - `Type::WARNING`: Warning alert.
+     * - `Type::INFO`: Info alert.
+     * - `Type::LIGHT`: Light alert.
+     * - `Type::DARK`: Dark alert.
+     *
+     * @return self A new instance of the current class with the specified alert type.
      */
-    public function dark(): self
+    public function type(Type $type): self
     {
-        return $this->addClassNames('alert-dark');
+        $new = clone $this;
+        Html::addCssClass($new->attributes, ['persistent' => self::CLASS_COMPONENT, 'alert-' . $type->value]);
+
+        return $new;
     }
 
     /**
-     * Render header tag
+     * Run the alert widget.
+     *
+     * @return string The HTML representation of the element.
      */
-    private function renderHeader(): ?string
+    public function render(): string
     {
-        if ($this->header === null) {
-            return null;
+        $attributes = $this->attributes;
+        $attributes['role'] = 'alert';
+        $content = '';
+        $id = $this->id;
+        $toggle = '';
+
+        Html::addCssClass($attributes, ['persistent' => 'alert']);
```

Comment:
```suggestion
        Html::addCssClass($attributes, ['persistent' => self::CLASS_COMPONENT]);
```

But... do we need to set it in different methods if it's always set to `alert` on rendering?

Reply:
If you don't use any bootstrap specific class, you can add a class in addClass, then it should always be added.